src/utils.ts

```diff
@@ -61,3 +63,76 @@ export const sortSelectOptionAsc = (options: Readonly<SelectOption[]>): SelectOp
 
 export const sortSelectOptionDesc = (options: Readonly<SelectOption[]>): SelectOption[] =>
   [...options].sort(selectOptionDescComparator)
+
+/**
+ * Convert a byte size into a more readable value.
+ * @param size The size to convert.
+ * @returns An object containing the value and the size unit.
+ */
+export const toReadableFileSize = (
+  size: number,
+  formatter?: (size: number) => number
+): ReadableFileSize =>
+  (['KB', 'MB', 'GB', 'TB'] as FileSizeUnit[]).reduce<ReadableFileSize>(
+    (acc: DeepReadonly<ReadableFileSize>, curr: FileSizeUnit) =>
+      acc.value < 1000
+        ? { value: acc.value, unit: acc.unit }
+        : { value: formatter ? formatter(acc.value / 1000) : acc.value / 1000, unit: curr },
+    { value: size, unit: 'B' }
+  )
+
+/**
+ * Check if the file has the given MIME type or extension
+ * @param file The file to check.
+ * @param acceptedFormat The accepted MIME type or extension of the file.
+ * @returns A boolean that indicates if the file match the MIME type or the extension.
+ */
+export const isFileTypeAccepted = (file: DeepReadonly<File>, acceptedFormat: string): boolean => {
+  const suffixMimeTypeRegex = /\/.*$/
+  const fileName = file.name.toLocaleLowerCase()
+  const mimeType = file.type.toLocaleLowerCase()
+  const baseMimeType = mimeType.replace(suffixMimeTypeRegex, '')
+
+  const type = acceptedFormat.trim().toLocaleLowerCase()
+  return (
+    !type ||
+    mimeType === type ||
+    (type.endsWith('/*') && baseMimeType === type.replace(suffixMimeTypeRegex, '')) ||
+    (type.startsWith('.') && fileName.endsWith(type))
+  )
+}
+
+/**
+ * Check if each file has one of the MIME types or extensions.
+ * @param files The files to check.
+ * @param acceptedFormats The accepted formats (MIME type or extension) of the files.
+ * @returns A boolean that indicates if the files matches the MIME types or extensions.
+ */
+export const areFilesAccepted = (
+  files: DeepReadonly<File[]>,
+  acceptedFormats?: Readonly<string[]>
+): boolean =>
+  files.every(
+    (file: DeepReadonly<File>) =>
+      !acceptedFormats ||
+      !acceptedFormats.length ||
+      acceptedFormats.some((format: string) => isFileTypeAccepted(file, format))
+  )
+
+/**
+ * Convert a list of `File` into a list of `StoreFilesPayload` to store in the redux store.
+ * @param files The list of file to convert
+ * @returns A list of converted files.
+ */
+export const toStoreFilesPayload = (
```

Comment:
IMHO, this method should not be part of `utils` as it exposes the way the file is mapped into a command payload, this is the domain responsibility.
I would have moved the whole method into the controller itself and thus, we could implement later a `builder` in the `File` domain that could do this job.